warrior/WarriorCore/testcase_steps_execution.py

```diff
@@ -9,239 +10,349 @@
+            return self._execute_runmode_step(runmode_timer, runmode, self.step_status, value)
 
         elif retry_type is not None:
-            if retry_type.upper() == 'IF':
+            return self._execute_retry_type_step(retry_type, retry_cond, retry_cond_value,
+                                                 retry_interval, retry_value)
+        else:
+            return self._execute_step_otherwise(self.step_status)
+
+    def _report_step_as_not_run(self):
+        """
+        This function handles reporting of a step as not run.
+        """
+        keyword = self.current_step.get('Keyword')
+        kw_resultfile = step_driver.get_keyword_resultfile(self.data_repository, self.system_name,
+                                                           self.next_step, keyword)
+        Utils.config_Utils.set_resultfile(kw_resultfile)
+        Utils.testcase_Utils.pKeyword(keyword, self.current_step.get('Driver'))
+        Utils.testcase_Utils.reportStatus('Skip')
+        self.kw_resultfile_list.append(kw_resultfile)
+        self.data_repository['wt_junit_object'].update_count("skipped", "1", "tc",
+                                                             self.data_repository['wt_tc_timestamp'])
+        self.data_repository['wt_junit_object'].update_count("keywords", "1", "tc",
+                                                             self.data_repository['wt_tc_timestamp'])
+        kw_start_time = Utils.datetime_utils.get_current_timestamp()
+        step_impact = Utils.testcase_Utils.get_impact_from_xmlfile(self.current_step)
+        impact_dict = {"IMPACT": "Impact", "NOIMPACT": "No Impact"}
+        self.data_repository['wt_junit_object'].add_keyword_result(
+            self.data_repository['wt_tc_timestamp'], self.next_step, keyword,
+            "SKIPPED", kw_start_time, "0", "skipped",
+            impact_dict.get(step_impact.upper()), "N/A")
+        self.data_repository['step_{}_result'.format(self.next_step)] = "SKIPPED"
+        self.go_to_step_number = False
+        if self.current_triggered_action.upper() in ['ABORT', 'ABORT_AS_ERROR']:
+            return self.next_step, self.go_to_step_number, "break"
+        elif self.current_triggered_action.upper() in ['SKIP', 'NEXT']:
+            return self.next_step, self.go_to_step_number, "continue"
+        elif self.current_triggered_action == "SKIP_INVOKED":
+            if self.skip_invoked:
+                print_info("Skipping this step as it is an invoked step.")
+                return self.next_step, self.go_to_step_number, "continue"
+        # when 'onError:goto' value is less than the current step num,
+        # change the next iteration point to goto value
+        elif self.current_triggered_action and int(self.current_triggered_action) < self.next_step:
+            self.next_step = int(self.current_triggered_action) - 1
+        return self.next_step, self.go_to_step_number, "continue"
+
+    def _execute_step_on_goto(self):
```

Comment:
_execute_step_on_goto actually executes the current step or if the go_to_step_num matches the current step num

for easy to read/locate puposes can we rename this function to match the functionality somthing like execute_current_step

Reply:
Updated.